data_model/src/expression.rs

```diff
@@ -338,962 +499,561 @@ impl ContextValue {
 }
 
 impl From<ContextValue> for ExpressionBox {
+    #[inline]
     fn from(expression: ContextValue) -> Self {
         Expression::ContextValue(expression).into()
     }
 }
 
-/// Evaluates to the multiplication of right and left expressions.
-/// Works only for [`Value::U32`]
-#[derive(
-    Debug,
-    Display,
-    Clone,
-    PartialEq,
-    Eq,
-    Decode,
-    Encode,
-    Deserialize,
-    Serialize,
-    IntoSchema,
-    PartialOrd,
-    Ord,
-)]
-#[display(
+gen_expr_and_impls! {
+    /// Evaluates to the multiplication of right and left expressions.
+    /// Works only for [`Value::U32`]
+    #[derive(
+        Debug,
+        Display,
+        Clone,
+        PartialEq,
+        Eq,
+        Decode,
+        Encode,
+        Deserialize,
+        Serialize,
+        IntoSchema,
+        PartialOrd,
+        Ord,
+    )]
+    #[display(
     fmt = "{}*{}", // Keep without spaces
-    "left.parenthesise(Operation::Multiply)",
-    "right.parenthesise(Operation::Multiply)"
-)]
-pub struct Multiply {
-    /// Left operand.
-    pub left: EvaluatesTo<u32>,
-    /// Right operand.
-    pub right: EvaluatesTo<u32>,
+        "self.left.parenthesise(Operation::Multiply)",
+        "self.right.parenthesise(Operation::Multiply)"
+    )]
+    pub Multiply(u32, u32) -> u32
+}
+
+gen_expr_and_impls! {
+    /// Evaluates to the division of right and left expressions.
```

Comment:
I think you should specify that the left is divided by the right. E.g. you can say which expression is the dividend and which is the divisor